ProcessLib/HydroMechanics/HydroMechanicsFEM.h

```diff
@@ -0,0 +1,425 @@
+                pressure_size> const>(local_xdot.data() + pressure_index,
+                                      pressure_size);
+        auto u_dot = Eigen::Map<typename ShapeMatricesType::template VectorType<
+            displacement_size> const>(local_xdot.data() + displacement_index,
+                                      displacement_size);
+
+        auto local_Jac = MathLib::createZeroedMatrix<
+            typename ShapeMatricesType::template MatrixType<
+                displacement_size + pressure_size,
+                displacement_size + pressure_size>>(
+            local_Jac_data, displacement_size + pressure_size,
+            displacement_size + pressure_size);
+
+        auto local_rhs = MathLib::createZeroedVector<
+            typename ShapeMatricesType::template VectorType<displacement_size +
+                                                            pressure_size>>(
+            local_rhs_data, displacement_size + pressure_size);
+
+        typename ShapeMatricesTypePressure::NodalMatrixType laplace_p;
+        laplace_p.setZero(pressure_size, pressure_size);
+
+        typename ShapeMatricesTypePressure::NodalMatrixType storage_p;
+        storage_p.setZero(pressure_size, pressure_size);
+
+        typename ShapeMatricesType::template MatrixType<displacement_size,
+                                                        pressure_size>
+            Kup;
+        Kup.setZero(displacement_size, pressure_size);
+
+        double const& dt = _process_data.dt;
+
+        SpatialPosition x_position;
+        x_position.setElementID(_element.getID());
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            x_position.setIntegrationPoint(ip);
+            auto const& w = _ip_data[ip].integration_weight;
+
+            auto const& N_p = _ip_data[ip]._N_p;
+            auto const& dNdx_p = _ip_data[ip]._dNdx_p;
+
+            auto const& B = _ip_data[ip]._b_matrices;
+            auto const& sigma = _ip_data[ip]._sigma;
+
+            auto const& C = _ip_data[ip]._C;
+
+            double const S =
+                _process_data.storage_coefficient(t, x_position)[0];
+            double const K_over_mu =
+                _process_data.intrinsic_permeability(t, x_position)[0] /
+                _process_data.fluid_viscosity(t, x_position)[0];
+            auto const alpha = _process_data.biot_coefficient(t, x_position)[0];
+            auto const rho_sr = _process_data.solid_density(t, x_position)[0];
+            auto const rho_fr = _process_data.fluid_density(t, x_position)[0];
+            auto const porosity = _process_data.porosity(t, x_position)[0];
+            auto const body_force =
+                _process_data.specific_body_force(t, x_position);
+            assert(body_force.size() == DisplacementDim);
+            auto const b =
```

Comment:
is it necessary to evaluate the body force at each gauss point and each time step?